### PR TITLE
rewrote Configuring Lookback section

### PR DESCRIPTION
**Pull request by tanberry**
Rewrote the "Configuring Lookback" section in the Readme to explain new lookback behaviour when "statefulness" is implemented. Also added a link to the "Running as non-root" section in the Kubernetes.md that explains how set it up for non-root.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ By default the agent will run as root. Below are environment-specific instructio
 * [Running as Non-Root on Kubernetes](KUBERNETES.md#run-as-non-root)
 * [Running as Non-Root on OpenShift](OPENSHIFT.md#run-as-non-root)
 
+If you configure the agent to run as non-root, review the [documentation about enabling "statefulness" for the agent](KUBERNETES.md#enabling-persistent-agent-state).
```
Comment (jorgebay) [inline docs/README.md:67]: `KUBERNETES.md#enabling-persistent-agent-state` points to a location that doesn't exist in the markdown file. Was the idea to also modify the `KUBERNETES.md` file?

File: docs/README.md [hunk 2]
```diff
@@ -157,12 +159,23 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
 
 ### Configuring Lookback
 
-The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK`. The set of valid values for this option are:
+The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK`.
```
Comment (jorgebay) [inline docs/README.md:162]:
```suggestion
The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK` variable.
```

File: docs/README.md [hunk 3]
```diff
@@ -157,12 +159,23 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...
+
+The valid values for this option are:
+   * When set to **`none`**: lookback is disabled, and LogDNA continues to read from where the file ended at the time of the stop/restart action.
```
Comment (jorgebay) [inline docs/README.md:165]: In technical terms `none` means the Agent will go to the end of the file, ignoring previously written lines, and it will read new lines of the file as those are added (tailing).

```suggestion
   * When set to **`none`**: lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were wrote before the time the Agent restarted.
```

File: docs/README.md [hunk 4]
```diff
@@ -157,12 +159,23 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...

+       * If the file is not present in the “state file” and the file is less than 8KiB, start at the beginning. If the file is larger than 8KiB, start at the end.

+   * When set to **`start`**:
+    * If there is information in the “state file”, use the last recorded state.

+    * If the file is not present in the “state file”, start at the beginning.

+
+The state file location is defined using the LOGDNA_DB_PATH env variable (the default path is /var/lib/logdna/agent_state.db).
```
Comment (jorgebay) [inline docs/README.md:173]:
```suggestion
The state file location is defined using the `LOGDNA_DB_PATH` env variable (the default path is /var/lib/logdna/agent_state.db).
```

File: docs/README.md [hunk 5]
```diff
@@ -157,12 +159,23 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...
+
+**Notes:**
+* If you configure the agent to run as non-root, review the [documentation](KUBERNETES.md#enabling-persistent-agent-state) about enabling "statefulness" for the agent.
+* When upgrading from LogDNA Agent versin 3.0 to 3.1, the state file will initially be empty, so the lookback setting will be used for existing files. After that (i.e. on process restart), the state file will be present and will be used.
```
Comment (jorgebay) [inline docs/README.md:177]: version :)

**Review by jorgebay — COMMENTED** [review]
Nice detailed info on lookback!

I've added some comments/suggestions below.

File: docs/README.md [hunk 6]
```diff
@@ -157,12 +159,23 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...
+
+The valid values for this option are:
+   * When set to **`none`**: lookback is disabled, and LogDNA continues to read from where the file ended at the time of the stop/restart action.

+   * When set to **`smallfiles`** (default):
+       * If there is information in the “state file”, use the last recorded state.

+       * If the file is not present in the “state file” and the file is less than 8KiB, start at the beginning. If the file is larger than 8KiB, start at the end.

+   * When set to **`start`**:
```
Comment (jorgebay) [inline docs/README.md:169]: hm... in the github markdown viewer, this is shown as `start: *` ...

File: docs/README.md (hunk 1 above)
Comment (jorgebay) [reply docs/README.md:67]: oh, I see that it relates to #109 , nvm.

File: docs/README.md (hunk 3 above)
Comment (tanberry) [reply docs/README.md:165]: Ah, thanks much ore clear. Two grammar changes though please:

"... Agent will start reading new..." 
"...lines that were written before..."

File: docs/README.md (hunk 5 above)
Comment (tanberry) [reply docs/README.md:177]: oh my gosh my horrible typing! Thank you for catching that. I will fix and add to the PR.

**Comment by tanberry** [comment]
@jorgebay thank you for the edits! It is ready to merge but I guess we wait until the 3.1 release goes out, and for @c-nixon  and @alfeng6  to review.

File: docs/README.md [hunk 7]
```diff
@@ -157,12 +159,24 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...
+      * lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were written before the time the Agent restarted.
+   * When set to **`smallfiles`** (default):
+       * If there is information in the “state file”, use the last recorded state.

+       * If the file is not present in the “state file” and the file is less than 8KiB, start at the beginning. If the file is larger than 8KiB, start at the end.

+   * When set to **`start`**:
+    * If there is information in the “state file”, use the last recorded state.

+    * If the file is not present in the “state file”, start at the beginning.
```
Comment (jorgebay) [inline docs/README.md:172]: I think the indentation was messing up the sublist.

```suggestion
       * If there is information in the “state file”, use the last recorded state. 
       * If the file is not present in the “state file”, start at the beginning. 
```

File: docs/README.md (hunk 1 above)
Comment (alfeng6) [inline docs/README.md:67]:
```suggestion
If you configure the LogDNA Agent to run as non-root, review the [documentation about enabling "statefulness" for the agent](KUBERNETES.md#enabling-persistent-agent-state).
```

File: docs/README.md [hunk 8]
```diff
@@ -157,12 +159,24 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
 
 ### Configuring Lookback
 
-The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK`. The set of valid values for this option are:
+The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by the `LOGDNA_LOOKBACK` variable.
+
+The valid values for this option are:
+   * When set to **`none`**:
+      * lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were written before the time the Agent restarted.
```
Comment (alfeng6) [inline docs/README.md:166]:
```suggestion
      * lookback is disabled, and LogDNA Agent will read new lines as those are added to the file, ignoring the lines that were written before the time the Agent restarted.
```

Comment (alfeng6) [inline docs/README.md:166]:
```suggestion
      * lookback is disabled, and LogDNA Agent will reading new lines as those are added to the file, ignoring the lines that were written before the time the LogDNA Agent restarted.
```

File: docs/README.md [hunk 9]
```diff
@@ -157,12 +159,24 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...

+       * If the file is not present in the “state file” and the file is less than 8KiB, start at the beginning. If the file is larger than 8KiB, start at the end.

+   * When set to **`start`**:
+    * If there is information in the “state file”, use the last recorded state.

+    * If the file is not present in the “state file”, start at the beginning.

+
+The state file location is defined using the `LOGDNA_DB_PATH` env variable (the default path is /var/lib/logdna/agent_state.db).
```
Comment (alfeng6) [inline docs/README.md:174]:
```suggestion
The state file location is defined using the `LOGDNA_DB_PATH` env variable (the default path is `/var/lib/logdna/agent_state.db`).
```

File: docs/README.md [hunk 10]
```diff
@@ -157,12 +159,24 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
...
+
+**Notes:**
+* If you configure the agent to run as non-root, review the [documentation](KUBERNETES.md#enabling-persistent-agent-state) about enabling "statefulness" for the agent.
```
Comment (alfeng6) [inline docs/README.md:177]:
```suggestion
* If you configure the LogDNA Agent to run as non-root, review the [documentation](KUBERNETES.md#enabling-persistent-agent-state) about enabling "statefulness" for the LogDNA Agent.
```

**Review by alfeng6 — DISMISSED** [review]
looks good, a few small changes

File: docs/README.md (hunk 6 above)
Comment (tanberry) [reply docs/README.md:170]: fixedit,thanks

File: docs/README.md (hunk 7 above)
Comment (tanberry) [reply docs/README.md:172]: Hmmm let me try again, becuse I need those to be level-two bullets, so mor eindented.

Comment (jorgebay) [reply docs/README.md:174]: This issue persists after the latest changes.

File: docs/README.md [hunk 11]
```diff
@@ -157,12 +159,24 @@ Check out [Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-dat
 
 ### Configuring Lookback
 
-The lookback strategy determines how the agent handles existing files on startup. This strategy is determined by `LOGDNA_LOOKBACK`. The set of valid values for this option are:
+The lookback strategy determines how the agent handles existing files on agent startup. This strategy is determined by the `LOGDNA_LOOKBACK` variable.
+
+By default, the agent provides a "stateful", or persistent, collection of files that can be referenced whenever the agent is restarted, in order to return (or look back) to see any files that were ingested during the time that the agent was not running. The state file location is defined using the `LOGDNA_DB_PATH` environment variable in the YAML file (the default path is /var/lib/logdna/agent_state.db).
```
Comment (jorgebay) [inline docs/README.md:164]: `LOGDNA_DB_PATH` setting should point to a directory.

```suggestion
By default, the agent provides a "stateful", or persistent, collection of files that can be referenced whenever the agent is restarted, in order to return (or look back) to see any files that were ingested during the time that the agent was not running. The state directory location is defined using the `LOGDNA_DB_PATH` environment variable in the YAML file (the default path is `/var/lib/logdna`).
```

**Comment by jorgebay** [comment]
@tanberry let me know if I can help w/ the final formatting issues.

**Review by alfeng6 — APPROVED** [review]
looks great

File: docs/README.md (hunk 6 above)
Comment (jorgebay) [reply docs/README.md:174]: There's still an issue with indentation of the list, that is shown in the viewer:


Screenshot 2021-03-31 at 09 40 27

**Review by jorgebay — APPROVED** [review]
